server/plugin.go

```diff
@@ -68,12 +69,17 @@ func (p *Plugin) OnActivate() error {
 	store := store.NewStore(mm, conf)
 	proxy := proxy.NewProxy(mm, awsClient, conf, store)
 
+	mutex, err := cluster.NewMutex(p.API, "PP_Cluster_Mutex")
+	if err != nil {
+		return errors.Wrapf(err, "failed creating cluster mutex")
+	}
+
 	p.api = &api.Service{
 		Mattermost:   mm,
 		Configurator: conf,
 		Proxy:        proxy,
 		AppServices:  appservices.NewAppServices(mm, conf, store),
-		Admin:        admin.NewAdmin(mm, conf, store, proxy),
+		Admin:        admin.NewAdmin(mm, conf, store, proxy, awsClient, mutex),
```

Comment:
nit: 0/5 is there a strong reason to create cluster.Mutex in plugin.go rather than in the admin service itself?

Reply:
No strong reason, but mutex needs `p.API` for creation. I did not wanted to drag this field in the `Admin` as we already have the `pluginapi.client` there.